pkg/controller/machineset/machineset_controller.go

```diff
@@ -230,12 +245,188 @@ func (c *Controller) processNextWorkItem() bool {
+	if true && ms.DeletionTimestamp == nil {
+		manageMachinesErr = c.manageMachines(filteredMachines, ms)
+	}
+
+	if manageMachinesErr != nil {
+		return manageMachinesErr
+	}
+
+	return nil
+}
+
+func (c *Controller) manageMachines(machines []*clusteroperator.Machine, ms *clusteroperator.MachineSet) error {
+	/*machineSetKey, err := controller.KeyFunc(ms)
+	if err != nil {
+		// FIXME
+		return nil
+	}*/
+
+	machinePrefix := ms.ObjectMeta.Name + "-"
+	//c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ machine prefix: %v", machinePrefix)
+
+	machineSetMachines := []*clusteroperator.Machine{}
+	//machinesToCreate := []*clusteroperator.Machine{}
+	machinesToDelete := []*clusteroperator.Machine{}
+	//machinesToUpdate := []*clusteroperator.Machine{}
+
+	deleteCount := int(0)
+	createCount := int(0)
+
+
+	// caculate how many to create/delete
+	if len(machines) > ms.Spec.Size {
+		deleteCount = len(machines) - ms.Spec.Size
+	}
+	if len(machines) < ms.Spec.Size {
+		createCount = ms.Spec.Size - len(machines)
+	}
+
+	for _, machine := range machines {
+		if ! strings.HasPrefix(machine.ObjectMeta.Name, machinePrefix) {
+			// poorly-named
+			c.logger.WithField("machineset", ms.ObjectMeta.Name).Infof("JDIAZ child machine without expected prefix %v", machine.ObjectMeta.Name)
+			machinesToDelete = append(machinesToDelete, machine)
+			createCount++
+			deleteCount++
+			continue
+		}
+
+		if currentMachineNeedsUpdate(machine, ms) {
+			c.logger.WithField("machineset", ms.ObjectMeta.Name).Errorf("JDIAZ child machine needs update %v", machine.ObjectMeta.Name)
+			machinesToDelete = append(machinesToDelete, machine)
+			createCount++
+			deleteCount++
+			continue
+		}
+
+		machineSetMachines = append(machineSetMachines, machine)
+
+	}
+
+	// Sync machines
+	machineConfig := buildNewMachine(ms, machinePrefix)
+	c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ machine template: %+v", machineConfig)
+
+	for i := 0 ; i < createCount ; i++ {
+		_, err := c.client.ClusteroperatorV1alpha1().Machines(ms.ObjectMeta.Namespace).Create(machineConfig)
+		if err != nil {
+			// FIXME add to error list?
+			c.logger.WithField("machineset", ms.ObjectMeta.Name).Error("JDIAZ error creating machine...")
+			// update expectations?
+		}
+	}
+
+	c.logger.WithField("machineset", ms.ObjectMeta.Name).Debug("JDIAZ delete count :v", deleteCount)
+	for _, m := range machinesToDelete {
+		c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ deleting machine %v", m.Name)
+		c.client.ClusteroperatorV1alpha1().Machines(ms.ObjectMeta.Namespace).Delete(m.Name, &metav1.DeleteOptions{})
+		// catch error ^^^
+		deleteCount--
+	}
+
+	var machineToDelete *clusteroperator.Machine
+	c.logger.WithField("machineset", ms.ObjectMeta.Name).Debugf("JDIAZ deleting %v more machines", deleteCount)
+	for i := 0 ; i < deleteCount ; i++ {
+		machineToDelete, machineSetMachines = machineSetMachines[0], machineSetMachines[1:]
```

Comment:
I don't see the point of overwriting the slice each time. Can't we simply use `machineSetMachines[i]` in each iteration?

`c.client.ClusteroperatorV1alpha1().Machines(ms.ObjectMeta.Namespace).Delete(machineSetMachines[i].Name, &metav1.DeleteOptions{})`